### ISSUE TITLE
Technical projects

### ISSUE THREAD

**Issue by cronvel** [ayojs/ayo#59]
Hello,

It would be nice to have a list of technical projects for Ayo.
Some people may want to contribute, but without goals, priorities and aimed features, it's hard to jump in the bandwagon.

From what I read on other issues, there is at least:
* ES6 module
* webworker-like thread

I may suggest two new things:
* core http2
* core websocket

I think the core value of Node/Ayo is the web, and IMHO it should come *battery-included* with all basic web stuff. The existing [*ws* module](https://www.npmjs.com/package/ws) has many years of maturity, I think it would be nice if it find its way to the core.

Anyway, having the technical todo list could help undecided people to make the move and pick one item.

**Comment by addaleax** [comment]
> core http2

Just so you know, this fork currently includes Node’s experimental HTTP2 implementation, and I don’t see it being removed here.

**Comment by cronvel** [comment]
@addaleax great, I missed that one!

**Comment by valpackett** [comment]
I'd like to suggest some security things:

- replacing the weird bundled fork of openssl with dynamically linked (system) libtls/libcrypto;
- examining random number generation code – it probably still uses the openssl RNG which should not be used anywhere ever – if so, replacing it with what libsodium does

**Comment by addaleax** [comment]
@myfreeweb yeah, those were definitely thing that were brought up in Node.js and that might make sense but also require a ton of volunteer work. https://github.com/nodejs/node/pull/13575 is probably a really great start, though
